dima/react-movies/src/components/searchBar/searchBar.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import SearchFilter from "./searchFilter";
+import "./searchBar.css";
+import ErrorBoundary from "../errorBoundary";
+import * as QueryString from "query-string";
+import FilterProperty from "../../enums/FilterPropery";
+import { RouteComponentProps } from "react-router";
+
+interface SearchBarProps {
+  searchMovies: (searchBarValue: string, searchedBy: string) => void;
+  route: RouteComponentProps;
+  fetchMovies: () => void;
+  currentSortType: string;
+}
+
+interface SearchBarState {
+  searchBarValue: string;
+  searchedBy: string;
+}
+
+export default class SearchBar extends React.Component<
+  SearchBarProps,
+  SearchBarState
+> {
+  state: SearchBarState = {
+    searchBarValue: "",
+    searchedBy: FilterProperty.title,
+  };
+  onSearchBarChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { value: searchBarValue } = e.target;
+    this.setState({ searchBarValue });
+  };
+  toggleSearchCategory = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ): void => {
+    const { value } = e.currentTarget;
+    this.setState({ searchedBy: value.toLowerCase() });
+  };
+
+  onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { history } = this.props.route;
+    const { searchBarValue, searchedBy } = this.state;
+    const urlParams = {
+      searchBy: searchedBy,
+      search: searchBarValue,
+    };
+    history.push({
+      pathname: "/search",
+      search: QueryString.stringify(urlParams),
+    });
```

Comment:
Better not to do history push in SearchBar. SearchBar is a reusable component and theoretically might be used on other pages. History push is really hardcoded and specific logic that should be used on specific page not on the common component.